modules/dnn/include/opencv2/dnn/all_layers.hpp

```diff
@@ -556,6 +556,37 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
         static Ptr<Layer> create(const LayerParams& params);
     };
 
+    class CV_EXPORTS DataAugmentationLayer : public Layer
+    {
+    public:
+        static Ptr<DataAugmentationLayer> create(const LayerParams& params);
+    };
+
+    class CV_EXPORTS CorrelationLayer : public Layer
+    {
+    public:
+        static Ptr<CorrelationLayer> create(const LayerParams& params);
+    };
```

Comment:
With a patch from https://github.com/opencv/opencv/pull/16840 do we still need CorrelationLayer?

Reply:
Yes, we need it.CorrelationLayer from ONNX (CP-VTON model) needed MatMul inputs (#16840), but CorrelationLayer from Caffe based on another logic.

Reply:
Yes, sorry, just mixed with CP-VTON